AzureIoTHub.agent.lib.nut

```diff
@@ -624,6 +628,224 @@ class AzureIoTHub {
+                }
+            }.bindenv(this);
+
+            request.sendasync(onSent);
+        }
+
+        // If the device is already registered and assigned to an IoT Hub,
+        // this method returns Device Connection String via the onCompleted handler.
+        //
+        // Parameters:
+        //     onCompleted : Function       A function to be called when the operation is completed or an error occurs
+        //                                  The callback signature:
+        //                                  onCompleted(error, response, connectionString), where
+        //                                      error : Integer     0 if the operation is successful, otherwise an error code
+        //                                      response : Table    Key-value table with the response provided by Azure server. May be null.
+        //                                                          For information on the response format, please see the Azure documentation.
+        //                                                          May also contain error details
+        //                                      connectionString :  Device connection string. null in case of an error
+        //                                          String
+        //
+        // Returns:                         Nothing.
+        function getConnectionString(onCompleted) {
+            local sasExpTime = time() + AZURE_DPS_SAS_TTL;
+            local sas = AzureIoTHub.SharedAccessSignature(_resUri, AZURE_DPS_REG_KEY_NAME, _deviceKey, sasExpTime).toString();
+
+            _headers["Authorization"] = sas;
+
+            local url = format(AZURE_DPS_REG_STATUS_ENDPOINT_FMT, AZURE_DPS_GLOBAL_HOST, _scopeId, _regId, AZURE_DPS_API_VERSION);
+            local request = http.post(url, _headers, _regIdBody);
+
+            local onSent = function(resp) {
+                local body = _parseBody(resp.body);
+
+                if (resp.statuscode == AZURE_DPS_OK_CODE) {
+                    if (!("status" in body)) {
+                        server.error("Response body doesn't have the \"status\" field");
+                        onCompleted(AZURE_ERROR_GENERAL, body, null);
+                        return;
+                    }
+
+                    if (body["status"] == "assigned") {
+                        if ("assignedHub" in body) {
+                            onCompleted(0, body, _connectionString(body["assignedHub"]));
+                        } else {
+                            server.error("Response body doesn't have the \"assignedHub\" field");
+                            onCompleted(AZURE_ERROR_GENERAL, body, null);
+                        }
+                    } else {
+                        onCompleted(AZURE_DPS_ERROR_NOT_REGISTERED, body, null);
+                    }
+                } else if (resp.statuscode == AZURE_DPS_NOT_FOUND_CODE) {
+                    onCompleted(AZURE_DPS_ERROR_NOT_REGISTERED, body, null);
```

Comment:
We might have discussed this already. If so, please remind me. 

Q1: we are replacing Azure error with our internal one. What's the point?
Q2: are we documenting `AZURE_DPS_ERROR_NOT_REGISTERED` error anywhere?